core/src/main/java/org/geysermc/geyser/translator/text/MessageTranslator.java

```diff
@@ -412,7 +409,7 @@ public static String normalizeSpace(String string) {
                 whitespacesCount++;
             } else {
                 startWhitespaces = false;
-                newChars[count++] = (actualChar == CHAR_NON_BREAKING_SPACE ? CHAR_SPACE : actualChar);
+                newChars[count++] = (actualChar == '\u00A0' ? ' ' : actualChar);
```

Comment:
Just explain what the \u00A0 character is and why we're replacing it and you're set.